### ISSUE TITLE
[BUG] Compressions not working on Titan Xp (CC 6.1)

### ISSUE THREAD

**Issue by DrivenByNostalgia** [NVIDIA/nvcomp#98]
Hi everyone,

should nvCOMP be working for Pascal generation cards? Despite the docs mentioning that the minimum hardware requirement for nvCOMP was compute capabilities 6.0 ("Pascal (sm60) or higher GPU architecture is required. Volta (sm70)+ GPU architecture is recommended for best results."), the compression fails on all our systems with Titan Xp cards, which have compute capability 6.1. On newer cards, all compression modes work as desired. We use nvCOMP in version nvcomp_3.0.5_windows_12.x.

We observe three different behaviors:
1) BitcompManager::get_compressed_output_size crashes the application with: "Unhandled exception at 0x00007FFD07BB286E (ucrtbase.dll) in *.exe: Fatal program exit requested.".
2) ZstdManager::compress throws the exception: "CUDA error: 98 (cudaErrorInvalidDeviceFunction: invalid device function)
  from: cudaOccupancyMaxActiveBlocksPerMultiprocessor( &num_blocks_per_sm, lz_compression_kernel, block_size, 0 )".
3) get_compressed_output_size with all other managers (ANS, Cascaded, Deflate, Gdeflate, LZ4, Snappy) returns 0 on any compression input.

In summary, none of the compressions work on any of our Titan Xp cards. We do not have any other Pascal cards for testing, so I cannot say whether the documented minimum requirement of compute capability is wrong and Pascal cards are not supported altogether or whether this is an issue specific to the Titan Xp. For the moment, we can circumvent the issue by simply deactivating compression on systems with CC < 7.0, but it would be sad to lose this feature for an entire GPU generation.

Best regards,
Daniel

**Comment by eschmidt-nvidia** [comment]
Hi Daniel,

Unfortunately, this is a bug in our documentation.

We depend on cuda::std::atomic, which only supports CC >= 7.0 on Windows (CC >= 6.0 on Unix). 

As a result, our Windows builds do not support CC < 7.0. 

-Eric

**Comment by DrivenByNostalgia** [comment]
Ok, thank you for clarifying that.
